OverlayDDU.h

```diff
@@ -257,26 +258,71 @@ class OverlayDDU : public Overlay
                 const float rpmPct = (rpm-lo) / (hi-lo);
 
                 const float ww = 0.16f;
-                for( int i=0; i<8; ++i )
+                if (!(ir_EngineWarnings.getInt() & irsdk_pitSpeedLimiter))
                 {
-                    const float lightPct = i/8.0f;
-                    const float lightRpm = lo + (hi-lo) * lightPct;
+                    for (int i = 0; i < 8; ++i)
+                    {
+                        const float lightPct = i / 8.0f;
+                        const float lightRpm = lo + (hi - lo) * lightPct;
 
-                    D2D1_ELLIPSE e = { float2(r2ax(0.5f-ww/2+(i+0.5f)*ww/8),r2ay(0.065f)), r2ax(0.007f), r2ax(0.007f) };
+                        D2D1_ELLIPSE e = { float2(r2ax(0.5f - ww / 2 + (i + 0.5f) * ww / 8),r2ay(0.065f)), r2ax(0.007f), r2ax(0.007f) };
 
-                    if( rpmPct < lightPct ) {
-                        m_brush->SetColor( outlineCol );
-                        m_renderTarget->DrawEllipse( &e, m_brush.Get() );
+                        if (rpmPct < lightPct) {
+                            m_brush->SetColor(outlineCol);
+                            m_renderTarget->DrawEllipse(&e, m_brush.Get());
+                        }
+                        else {
+                            if (lightRpm < ir_session.rpmSLFirst)
+                                m_brush->SetColor(float4(1, 1, 1, 1));
+                            else if (lightRpm < ir_session.rpmSLLast)
+                                m_brush->SetColor(warnCol);
+                            else
+                                m_brush->SetColor(float4(1, 0, 0, 1));
+                            m_renderTarget->FillEllipse(&e, m_brush.Get());
+                        }
+                    }
+                }
+                else
```

Comment:
Can't the logic in the else branch be rolled into the above loop? We really just need additional logic that figures out the color and whether to use DrawEllipse or FillEllipse, right?

Reply:
That is how I originally did it, but it created a small delay between each LED being drawn, creating almost a wave affect. I moved it outside of the loop to remove that small delay and now all ellipse are visually drawn together.

Reply:
I have just improved on this. I have put the pit limiter into its own for loop, not sure why I didn't do this in the first place. Feel free to test putting the limiter code into the main for loop, when I do it, I seem to get the pit limiter flashing right to left.